### PR TITLE
POTEL 51 - Cleanup OTel class loading

### PR DESCRIPTION
**Pull request by adinauer**
## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt [hunk 1]
```diff
@@ -741,43 +744,68 @@ class SentryAutoConfigurationTest {
...
-                assertThat(it).doesNotHaveBean(OpenTelemetryLinkErrorEventProcessor::class.java)
-                val options = it.getBean(SentryOptions::class.java)
-                assertThat(options.eventProcessors).noneMatch { processor -> processor.javaClass == OpenTelemetryLinkErrorEventProcessor::class.java }
+                println(SentryIntegrationPackageStorage.getInstance().integrations)
+                assertFalse(SentryIntegrationPackageStorage.getInstance().integrations.contains("SpringBoot3OpenTelemetryAgentWithoutAutoInit"))
+            }
+    }
+
+    @Test
+    fun `when AgentMarker is not on the classpath and auto init off, does not run SentryOpenTelemetryAgentWithoutAutoInitConfiguration`() {
```
Comment (lbloder) [inline sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt:767]: Not sure if we also need a `when AgentMarker is not on the classpath and auto init on, does not run SentryOpenTelemetryAgentWithoutAutoInitConfiguration` version of this?

File: sentry-samples/sentry-samples-spring-boot-jakarta/build.gradle.kts [hunk 2]
```diff
@@ -53,6 +53,9 @@ dependencies {
     implementation(projects.sentryGraphql22)
     implementation(projects.sentryQuartz)
     implementation(Config.Libs.OpenTelemetry.otelSdk)
+    implementation(projects.sentryOpentelemetry.sentryOpentelemetryAgentcustomization)
+//    implementation(projects.sentryOpentelemetry.sentryOpentelemetryBootstrap)
```
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-jakarta/build.gradle.kts:57]: updated in a follow up PR

File: sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt [hunk 3]
```diff
@@ -741,43 +744,68 @@ class SentryAutoConfigurationTest {
...
     @Test
-    fun `when OpenTelemetryLinkErrorEventProcessor is on the classpath and auto init off, creates OpenTelemetryLinkErrorEventProcessor`() {
+    fun `when AgentMarker is on the classpath and auto init off, runs SentryOpenTelemetryAgentWithoutAutoInitConfiguration`() {
+        SentryIntegrationPackageStorage.getInstance().clearStorage()
         contextRunner.withPropertyValues("sentry.dsn=http://key@localhost/proj", "sentry.auto-init=false")
             .run {
-                assertThat(it).hasSingleBean(OpenTelemetryLinkErrorEventProcessor::class.java)
-                val options = it.getBean(SentryOptions::class.java)
-                assertThat(options.eventProcessors).anyMatch { processor -> processor.javaClass == OpenTelemetryLinkErrorEventProcessor::class.java }
+                println(SentryIntegrationPackageStorage.getInstance().integrations)
```
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt:751]: updated in a follow up PR

File: sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt [hunk 4]
```diff
@@ -741,43 +744,68 @@ class SentryAutoConfigurationTest {
...
-    fun `when OpenTelemetryLinkErrorEventProcessor is on the classpath but auto init on, does not create OpenTelemetryLinkErrorEventProcessor`() {
-        contextRunner.withPropertyValues("sentry.dsn=http://key@localhost/proj", "sentry.auto-init=true")
+    fun `when AgentMarker is on the classpath and auto init on, does not run SentryOpenTelemetryAgentWithoutAutoInitConfiguration`() {
+        SentryIntegrationPackageStorage.getInstance().clearStorage()
+        contextRunner.withPropertyValues("sentry.dsn=http://key@localhost/proj")
             .run {
-                assertThat(it).doesNotHaveBean(OpenTelemetryLinkErrorEventProcessor::class.java)
-                val options = it.getBean(SentryOptions::class.java)
-                assertThat(options.eventProcessors).noneMatch { processor -> processor.javaClass == OpenTelemetryLinkErrorEventProcessor::class.java }
+                println(SentryIntegrationPackageStorage.getInstance().integrations)
```
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt:761]: updated in a follow up PR

File: sentry-opentelemetry/sentry-opentelemetry-agentcustomization/src/main/java/io/sentry/opentelemetry/SentryAutoConfigurationCustomizerProvider.java [hunk 5]
```diff
@@ -69,6 +64,15 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
         .addPropertiesSupplier(this::getDefaultProperties);
   }
 
+  private static void ensureSentryOtelStorageIsInitialized() {
+    /*
+    accessing Sentry.something will cause ScopesStorageFactory to run,
+    which causes OtelContextScopesStorage.init to register SentryContextStorage
+    as a wrapper. The wrapper can only be set until storage has been initialized by OpenTelemetry.
+    */
+    Sentry.getGlobalScope();
```
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-agentcustomization/src/main/java/io/sentry/opentelemetry/SentryAutoConfigurationCustomizerProvider.java:73]: I think the extracted method plus the comment here should suffice and I'd rather not clutter public API with a noop method.

File: sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt (hunk 1 above)
Comment (adinauer) [reply sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt:767]: addded in a follow up PR

**Review by lbloder — APPROVED** [review]
LGTM 👍

**Pull request merged**